tile_generator/templates/jobs/deploy-all.sh.erb

```diff
@@ -55,17 +55,22 @@ function import_opsmgr_variables() {
 	{% endif %}
 	{% endfor %}
 	{% endfor %}
+	<% empty_dict = {} %>
+	<% empty_list = [] %>
 	{% for release in context.releases.values() if release.consumes %}
 	{% for link in release.consumes %}
 	<% if_link('{{ link }}') do |link| %>
 	<% hosts = link.instances.map { |instance| instance.address } %>
-	export {{ link | upper }}_HOST="<%= link.instances[0].address %>"
+	export {{ link | upper }}_HOST="<%= link.instances.empty? ? ( puts "" ) : ( puts link.instances[0].address ) %>"
 	export {{ link | upper }}_HOSTS=<%= Shellwords.escape(hosts.to_json) %>
-    export {{ link | upper }}_PROPERTIES=<%= Shellwords.escape(link.properties.to_json) %>
+        export {{ link | upper }}_PROPERTIES=<%= Shellwords.escape(link.properties.to_json) %>
+        <% end.else do %>
+	export {{ link | upper }}_HOST=<%= Shellwords.escape(empty_list.to_json) %>
```

Comment:
I am confused on what is the intention on these few lines and why is it we are escaping empy lists and dicts?

Reply:
if_link is used when a link is found and regardless of 'optional' true or false,
end.else do will take of "unsetting" values when a link is missing.
Hence the "empty" list and dicts. I was not sure how to go about setting those to hardcoded empty [] and {} without running into trouble with multiple layers of tools handling and escaping contents. So I just used the existing pattern and empty lists and dicts..